api/proto/metrics/metrics.proto

```diff
@@ -173,6 +173,60 @@ message appCpuMetric {
   uint64 total_ns = 7;                 // like 'total' but in nanoseconds. Overflow may happen after ~40 years without EVE restart
 }
 
+message SmartAttr {
+  uint32 id = 1;
+  string name = 2;
+  uint64 value = 4; // the current value of the parameter
+  uint64 worst = 5; // worst value that Value has ever reached
+  uint64 thresh = 6;  // the value that the Value of the same attribute must reach for the attribute's state to be considered critical.
+  string when_failed = 9; // critical state if no empty
+  string raw_value = 10;
+}
+
+message ValueSelfTest {
+  uint32 num = 1;
+  string name = 2;
+  uint32 status = 3;
+  string remaining = 4;
+  uint64 lifetime = 5;  //in hours
+  string lba_of_first_error = 6;
+}
+
+message SmartMetric {
+  SmartAttr reallocated_sector_ct = 1;
+  SmartAttr power_on_hours = 2;
+  SmartAttr power_cycle_count = 3;
+  SmartAttr reallocated_event_count = 4;
+  SmartAttr current_pending_sector = 5;
+  bool needupdate = 6;
+}
+
+enum StorageStatus {
+  STORAGE_STATUS_UNSPECIFIED = 0;
+  STORAGE_STATUS_ONLINE = 1;   // The device or virtual device is in normal working order.
+  STORAGE_STATUS_DEGRADED = 2; // The virtual device has experienced a failure but can still function.
+  STORAGE_STATUS_FAULTED = 3;  // The device or virtual device is completely inaccessible.
+  STORAGE_STATUS_OFFLINE = 4;  // The device has been explicitly taken offline by the administrator.
+  STORAGE_STATUS_UNAVAIL = 5;  // The device or virtual device cannot be opened. In some cases, pools with UNAVAIL devices appear in DEGRADED mode.
+  STORAGE_STATUS_REMOVED = 6;  // The device was physically removed while the system was running.
+}
+
+message StorageDiskInfo {
+  string disk_name = 1;
+  StorageStatus status = 2;            // In ZFS
+  repeated SmartMetric smart_data = 3; // smartctl output
+}
+
+message StorageMetric {
```

Comment:
Is this per pool/mount point?
If so make it be repeated to plan for future, and have it include the mountpoint in StorageMetric

Reply:
I have added string with name for zpool